test/testdata/policy-controller/e2e/test-deployment-with-custom-resource.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: demo-scalable-custom-resource
+  name: test-deployment-custom-resource
+  labels:
+    app: demo-custom-resource
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: demo-custom-resource
+  template:
+    metadata:
+      labels:
+        app: demo-custom-resource
+    spec:
+      containers:
+          - args:
+              - resource-name="pods"
+          - name: demo-custom-resource
+          image: TEST_IMAGE
```

Comment:
The problem here is that the TEST_IMAGE as created has a full digest in it so it's harder to tell if it's been resolved to digest or not. I'd suggest that you have an image that has a tag, which would without this change be resolved to digest for a deployment, so creating a deployment with a tag here would be the simplest way to verify the functionality works. Then the resulting pods should have the digests resolved.

Reply:
Test was added.